### PR TITLE
Optional snapshot store spec

### PR DESCRIPTION
**Pull request by kelebra**
This PR partially addresses issue #24687. Port of `akka.persistence.OptionalSnapshotStoreSpec`.

This specification contains one test case (`"fail if PersistentActor tries to saveSnapshot without snapshot-store available"`) that fails for typed version because of one of the reasons below:

- Exception is not logged
- `akka.persistence.SaveSnapshotFailure` is not propagated back to typed internal protocol even though I see it being created on untyped part

Also it should be mentioned that in case logging validation is removed `stateProbe` actually receives instance of `State` just like nothing happened. Any help with this failed test is appreciated :sweat:

Two other tests were ported and work as expected.

### CODE REVIEW

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      message should include("No default snapshot store configured")
+      stateProbe.expectNoMessage()
+    }
+
+    "fail if PersistentActor tries to saveSnapshot without snapshot-store available" in {
+      val stateProbe = TestProbe[State]()
+      val log = logProbe(classOf[Logging.Error])
+      val persistentActor = spawn(persistentBehavior(stateProbe))
+      persistentActor ! AnyCommand
+      log.expectMessageType[Logging.Error].message.toString should include("No snapshot store configured")
```
Comment (patriknw) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala]: it's supposed to be logged here: https://github.com/akka/akka/blob/master/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala#L265

but `SnapshotterResponse` is only handled here: https://github.com/akka/akka/blob/master/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala#L182

try to add that case in the other running behavior also: https://github.com/akka/akka/blob/master/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala#L152

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      stateProbe.expectNoMessage()
```
Comment (patriknw) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala]: was this your second question? it's correct that it should receive the get the `State` here, failed snapshot is not fatal and it continues with the `andThen` callback anyway
(I think the snapshot saving is even completely async after the persisting of the event and probable after the `andThen` callback)

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala (hunk 1 above)
Comment (kelebra) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala]: Thanks! Reused `onSnapshotterResponse` in running behavior and updated expectations.

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala (hunk 2 above)
Comment (kelebra) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/OptionalSnapshotStoreSpec.scala]: Got it, updated expectation. Thanks for explanation.

**Comment by kelebra** [comment]
Now looks like this one is also done

**Comment by kelebra** [comment]
@patriknw, do you think anything else should be done in this PR?

**Comment by patriknw** [comment]
I fixed the merge conflict

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Comment by johanandren** [comment]
Refs #24687

**Pull request merged**